relay-metrics/src/protocol.rs

```diff
@@ -152,20 +152,41 @@ fn is_valid_name(name: &str) -> bool {
     false
 }
 
-/// Validates an MRI of the form `<ty>:<ns>/<name>@<unit>`
-///
-/// Note that the format used in the statsd protocol is different: Metric names are not prefixed
-/// with `<ty>:` as the type is somewhere else in the protocol.
-pub(crate) fn is_valid_mri(name: &str) -> bool {
-    let mut components = name.splitn(2, ':');
-    if components.next().is_none() {
-        return false;
-    }
+/// A metric name parsed as MRI, a naming scheme which includes most of the metric's bucket key
+/// (excl. timestamp and tags).
+pub struct MetricMri {
+    /// The metric type.
+    pub ty: MetricType,
+    /// The namespace/usecase for this metric. For example `sessions` or `transactions`.
+    pub namespace: String,
+    /// The actual name, such as `duration` as part of `d:transactions/duration@ms`
+    pub name: String,
+    /// The metric unit.
+    pub unit: MetricUnit,
```

Comment:
Now that the unit is in the MRI, it should be removed from the all other payloads as field.